### PR TITLE
Add SBI (Seðlabanki Íslands) provider

### LINKED ISSUE #234: Add SBI (Seðlabanki Íslands) provider
## Seðlabanki Íslands (Central Bank of Iceland)

- **Key:** SBI
- **Base currency:** ISK
- **Currencies:** ~20 vs ISK
- **API:** JSON/XML with Swagger docs, no auth
- **Publish schedule:** TBD

### Notes

Well-documented API with Swagger spec. ISK adds a Nordic currency not well covered by other providers. Tier 1 candidate.

### Checklist

See [NEW_PROVIDER.md](../blob/main/NEW_PROVIDER.md)

### PR DESCRIPTION
**Pull request by lineoffligbot**
## Summary

Closes #234

- Add `Providers::SBI` fetching daily exchange rates for 30+ currencies against the Icelandic króna (ISK) from the Central Bank of Iceland XML API
- Two GroupIDs combined per fetch: GroupID=9 (official reference, 10 currencies from 1981) and GroupID=7 (registered mid-rate, 22 currencies from 2006)
- Chunked backfill with `range: 365` to keep memory usage reasonable on the large historical dataset
- Add seed entry with publish_time 16 (UTC), weekdays only

## Test plan

- [x] VCR cassette recorded from live API (2026-03-24 to 2026-03-28)
- [x] Integration tests: fetch + import stores multiple currencies and dates
- [x] Parse unit tests: correct base/quote/rate extraction, GroupID=7 and GroupID=9
- [x] Edge cases: missing values, zero rates, unknown TimeSeries IDs
- [x] Full test suite passes (273 runs, 0 failures)
- [x] RuboCop clean (116 files, no offenses)

### CODE REVIEW

File: lib/providers/sbi.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    # GroupID=9 (official reference rates)
+    GROUP9_CURRENCIES = {
+      "4055" => "USD",
+      "4061" => "DKK",
+      "4064" => "EUR",
+      "4085" => "JPY",
+      "4088" => "CAD",
+      "4091" => "NOK",
+      "4097" => "XDR",
+      "4103" => "GBP",
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply lib/providers/sbi.rb:27]: XDR is intentionally in the mapping to document the full series list. Base#import handles the exclusion.

File: lib/providers/sbi.rb [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+      @dataset = parse(xml9, GROUP9_CURRENCIES) + parse(xml7, GROUP7_CURRENCIES)
+      self
+    rescue Net::OpenTimeout, Net::ReadTimeout, Socket::ResolutionError
+      @dataset = []
+      self
+    end
+
+    def parse(xml, currency_map = GROUP9_CURRENCIES.merge(GROUP7_CURRENCIES))
+      doc = Ox.load(xml, mode: :generic)
+      records = []
```
[the remark this answers is not in the document]
Comment (lineoffligbot) [reply lib/providers/sbi.rb:84]: parse is called twice per fetch (once per group), always with an explicit currency_map argument. The default is only there for standalone use in tests.
